src/Compile/CompilePanel.ts

```diff
@@ -132,13 +132,34 @@ export class CompilePanel {
   /**
    * Sets up an event listener to listen for messages passed from the webview context and
    * executes code based on the message that is recieved.
-   *
-   * @param webview A reference to the extension webview
-   * @param context A reference to the extension context
    */
   private _setWebviewMessageListener() {
-    /* NYI */
-    vscode.window.showInformationMessage('NYI');
+    const webview = this._panel.webview;
+    webview.onDidReceiveMessage((message: any) => {
+      const command = message.command;
+      const text = message.text;
+
+      switch (command) {
+        case 'compile-completed':  // Called when [Compile] button was pressed
```

Comment:
The `compile-completed` looks like a notification that occurs when compilation is completed. But in this switch phrase, it tries to compile here, so I'm a little confused.

(Q) When is this post message called? After compilation or Trying to compile?

Reply:
@jyoungyun,
[Compile] button in webview exits in HTML/JS sandbox.
When the button was clicked in HTML/JS sandbox, HTML/JS sandbox passes 'compile-completed' message to _extension_  (*.ts) sandbox, where the actual compilation by ONE can be done.

JS code that sends a message in this PR:
https://github.com/hyunsik-yoon/ONE-vscode/blob/31757c34798b70ffe1272f028ecddcd10e6e0f34/media/Compile/compile.js#L91

I think JS sandbox should handle UI related stuff. 
It's better to handle actual behavior in _extension_ sandbox because less security limitation and easier to debug with vscode.

Reply:
I'm sorry but I don't understand.

I saw the `TODO Run ONE compile command` under this function. So I think this request name would be better to be `request-compile` instead of `compile-completed`. Did I misunderstand what you were saying?

Reply:
`request-compile` would be better. I will rename.